src/survey/mod.rs

```diff
@@ -25,10 +25,10 @@ pub struct Parameters {
 impl Parameters {
```

Comment:
I am thinking in the background about ways to make the serialization a little less line oriented, if that makes sense.

You'd talked earlier about an AST kind of thing for the surveys, which is cool and I think has some interesting implications for this.